src/table.js

```diff
@@ -542,15 +551,79 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type) {
+  switch (type) {
+    case "string":
+      return typeof value === "string" || value == null
+        ? value
+        : String(value);
+    case "boolean":
+      if (typeof value === "string") {
+        const trimValue = value.trim();
+        return trimValue === "true" ? true : trimValue === "false" ? false : null;
```

Comment:
Just want to note that I was surprised that our boolean coercion returned `false` for a column of string values, but `true` for other types, such as a column of empty objects. _Maybe_ the default return value for strings should be `Boolean(value)`, but then that ruins detecting the strings "true" and "false". 
<img width="1065" alt="Screen Shot 2023-02-02 at 11 45 14 AM" src="https://user-images.githubusercontent.com/1373882/216387645-efc4e311-4ed2-499e-bee7-7de25a3bdb1f.png">

Reply:
It returned `null` for string values (rather than `false`). I'm not sure what the right response is when a Date, object, or other types are coerced to boolean. `TRUE` seems surprising to me, but if that's what we want to do, then it does seem strange to have type `string` alone not conform to that. Any thoughts, @mbostock?

Reply:
I think we have to do this because string is special: it’s the “untyped” format we use to serialize other types (namely in CSV).